symengine/parser.cpp

```diff
@@ -105,6 +108,15 @@ class ExpressionParser
             {"polygamma", polygamma},
             {"kronecker_delta", kronecker_delta}};
 
+    // maps string to corresponding double argument boolean function
+    std::map<std::string,
+             std::function<RCP<const Boolean>(const RCP<const Basic> &,
+                                              const RCP<const Basic> &)>>
+        double_arg_boolean_functions = {
```

Comment:
The compilation error is possibly due to the overloaded `Eq` function. See how I have done it for the `log` function above.